### DISCUSSION TITLE
Running CLI and loading model

### DISCUSSION THREAD

**Discussion by ronag** [SYSTRAN/faster-whisper#262] [Q&A] [answered]
Is the whole model loaded into memory every time whisper is run as a cli? Is there any way to avoid this so that a cli can be used as a batch processing tool?

**Comment by guillaumekln** [comment, marked as answer]
This project does not provide a CLI. Did you build your own CLI or are you using another project?

In general the CLI can accept multiple input files so that the model is loaded only once and used multiple times.

**Comment by Purfview** [comment]
> Is the whole model loaded into memory every time whisper is run as a cli? 

Yes.

> Is there any way to avoid this so that a cli can be used as a batch processing tool?

If args are implemented same as vanilla whisper then this should work:
`whisper.exe --model medium --language en "audio1" "audio2" "audio3"`
